### PR TITLE
Vector env

### PR DESCRIPTION
**Pull request by KornbergFresnel**

### CODE REVIEW

**Comment by KornbergFresnel** [comment]
@ziyuwan @VegeWong environment vectorization for simultaneous rollout. merged from malib-dev, and new environment wrapper for the registered environment. check this in `malib/envs`.

File: malib/agent/agent_interface.py [hunk 1]
```diff
@@ -310,6 +310,7 @@ def request_data(
 
         while batch is None:
             batch, info = ray.get(self._offline_dataset.sample.remote(buffer_desc))
+            # print(">>>>>> sample info:", info)
```
Comment (ziyuwan) [inline malib/agent/agent_interface.py:313]: maybe we should delete the debug output code, although you comment on it?

File: malib/algorithm/random/policy.py [hunk 2]
```diff
@@ -25,23 +27,33 @@ def __init__(
...
+        action_prob = torch.zeros((len(observation), self.action_space.n))
+        if "legal_moves" in kwargs:
+            mask = torch.zeros_like(logits)
+            mask[kwargs["legal_moves"]] = 1
+        elif "action_mask" in kwargs:
+            mask = torch.FloatTensor(kwargs["action_mask"])
+        else:
+            mask = torch.ones_like(logits)
+        logits = mask * logits
+        action = logits.argmax(dim=-1).view((-1, 1)).squeeze(-1).numpy()
```
Comment (ziyuwan) [inline malib/algorithm/random/policy.py:55]: use `argmax` here makes it doesn't act as random policy.
PS: should it also handle for Non-Discrete action space?

File: malib/envs/agent_interface.py [hunk 3]
```diff
@@ -128,7 +126,7 @@ def set_behavior_dist(self, distribution: Dict[PolicyID, float]) -> None:
             assert pid in self.policies, (pid, list(self.policies.keys()))
 
         # reset the sample distribution with 0.0 for all policies
```
Comment (ziyuwan) [inline malib/envs/agent_interface.py:128]: this comment should be changed

File: malib/agent/agent_interface.py (hunk 1 above)
Comment (KornbergFresnel) [reply malib/agent/agent_interface.py:313]: sure

File: malib/algorithm/random/policy.py (hunk 2 above)
Comment (KornbergFresnel) [reply malib/algorithm/random/policy.py:55]: the random policy is for no-RL-algorithm-involved tests; I think discrete action is enough.

**Pull request merged**